### PR TITLE
various add-ons: More example alerts

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOGS > Added change notes.
- Scan rules > Added example alert functionality.
- UnitTests > Added tests for example alerts.

## Related Issues
- zaproxy/zaproxy#6119
- zaproxy/zaproxy#8262

## Checklist
- [NA] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - The following scan rules now include example alert functionality for documentation generation purposes (Issue 6119):
     - HTTPS Content Available via HTTP
     - Remote Code Execution - Shell Shock (it now also uses Alert Refs (Issue 7100))
+    - Backup File Disclosure
```
Comment (thc202) [inline addOns/ascanrulesBeta/CHANGELOG.md:19]: Needs to be moved.

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/ascanrulesBeta/CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,8 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-
+### Changed
+- The Backup File Disclosure scan rule now includes example alert functionality for documentation generation purposes (Issue 6119):
```
Comment (ricekot) [inline addOns/ascanrulesBeta/CHANGELOG.md:8]: Remove the trailing colon?

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java [hunk 3]
```diff
@@ -765,21 +776,9 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
...
-                            .setOtherInfo(originalMessage.getRequestHeader().getURI().toString())
-                            .setSolution(
-                                    Constant.messages.getString(
-                                            "ascanbeta.backupfiledisclosure.soln"))
-                            .setEvidence(
-                                    Constant.messages.getString(
-                                            "ascanbeta.backupfiledisclosure.evidence",
-                                            originalURI,
-                                            candidateBackupFileURI))
+                    buildAlert(
```
Comment (ricekot) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java:779]: Is the `setName` call not needed?

Comment (thc202) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java:779]: It's the same as `getName()` which is set by default.

**Comment by kingthorin** [comment]
Tweaked

**Review by ricekot — APPROVED** [review]
Thanks!

**Pull request merged**